### PR TITLE
Add e2e test for volume metrics

### PR DESCRIPTION
**Pull request by gnufied**
Adding e2e test for volume metrics. These tests need to be run
in serial to verify accuracy

Fixes https://github.com/kubernetes/kubernetes/issues/52458

cc @saad-ali @jdumars 

/sig storage

### CODE REVIEW

File: test/e2e/storage/volume_metrics.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+		if err != nil {
+			framework.Failf("Error creating metrics grabber : %v", err)
+		}
+	})
+
+	It("should create prometheus metrics for volume provisioning and attach/detach", func() {
+		var err error
+
+		controlerMetrics, err := metricsGrabber.GrabFromControllerManager()
```
Comment (jeffvance) [inline test/e2e/storage/volume_metrics.go]: typo? controllerMetrics ?

File: test/e2e/storage/volume_metrics.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		if err != nil {
```
Comment (jeffvance) [inline test/e2e/storage/volume_metrics.go]: It's a bit more common to use `Expect` for this type of err  check. Though `Failf` is still seen...

File: test/e2e/storage/volume_metrics.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		claims := []*v1.PersistentVolumeClaim{pvc}
+
+		pod := framework.MakePod(ns, claims, false, "")
+		pod, err = c.CoreV1().Pods(ns).Create(pod)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = framework.WaitForPodRunningInNamespace(c, pod)
+
+		if err != nil {
+			framework.Failf("Error starting pod : %v", err)
```
Comment (jeffvance) [inline test/e2e/storage/volume_metrics.go]: ditto on `Expect`

File: test/e2e/storage/volume_metrics.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		}
+
+		framework.Logf("Deleting pod %q/%q", pod.Namespace, pod.Name)
+		framework.ExpectNoError(framework.DeletePodWithWait(f, c, pod))
+
+		updatedMetrics, err := metricsGrabber.GrabFromControllerManager()
+		if err != nil {
```
Comment (jeffvance) [inline test/e2e/storage/volume_metrics.go]: ditto on Expect

File: test/e2e/storage/volume_metrics.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+		for _, volumeOp := range volumeOperations {
+			verifyMetricCount(storageOpMetrics, updatedStorageMetrics, volumeOp)
+		}
+	})
+})
+
+func verifyMetricCount(oldMetrics map[string]int64, newMetrics map[string]int64, metricName string) {
+	oldCount, ok := oldMetrics[metricName]
+
+	if !ok {
```
Comment (jeffvance) [inline test/e2e/storage/volume_metrics.go]: could use `Expect(ok).To(BeTrue())`

File: test/e2e/storage/volume_metrics.go [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+		framework.Failf("Error getting metrics for %s", metricName)
+		return
+	}
+
+	newCount, ok := newMetrics[metricName]
+
+	if !ok {
```
Comment (jeffvance) [inline test/e2e/storage/volume_metrics.go]: ditto

**Comment by gnufied** [comment]
@jeffvance addressed all your comments. PTAL

**Comment by ericchiang** [comment]
Heads up that since being added this test has been flaking https://github.com/kubernetes/kubernetes/issues/52871

**Pull request merged**